web/src/client/dbus.js

```diff
@@ -70,9 +70,20 @@ import cockpit from "../lib/cockpit";
 class DBusClient {
   /**
    * @param {string} service - service name
+   * @param {string|undefined} address - D-Bus address. If no address is given, it connects
+   *   to the "system" bus.
    */
-  constructor(service) {
-    this.client = cockpit.dbus(service, { bus: "system", superuser: "try" });
+  constructor(service, address = undefined) {
+    const options = {
+      bus: address === undefined ? "system" : "none",
+      superuser: "try"
+    };
+
+    if (address) {
+      options.address = address;
```

Comment:
NP: I would find easier to read if there is no ternary operator and instead here it will be on separate line to overwrite bus to none.